kernel/src/engine/arrow_utils.rs

```diff
@@ -757,6 +762,58 @@ fn reorder_list<O: OffsetSizeTrait>(
     }
 }
 
+/// Arrow lacks the functionality to json-parse a string column into a struct column -- even tho the
+/// JSON file reader does exactly the same thing. This function is a hack to work around that gap.
+pub(crate) fn parse_json(
+    json_strings: Box<dyn EngineData>,
+    schema: SchemaRef,
+) -> DeltaResult<Box<dyn EngineData>> {
+    let json_strings: RecordBatch = ArrowEngineData::try_from_engine_data(json_strings)?.into();
+    let struct_array: StructArray = json_strings.into();
+    let json_strings = struct_array
```

Comment:
```suggestion
    let json_strings = json_strings
```

I think maybe the struct array conversion was a holdover from before. Not needed anymore.